libs/structwalk/walktype.go

```diff
@@ -27,7 +27,9 @@ import (
 //
 // ******************************************************************************************************
 
-type VisitTypeFunc func(path *structpath.PathNode, typ reflect.Type)
+type VisitTypeFunc func(path *structpath.PathNode, typ reflect.Type) error
```

Comment:
Can this be simpler? We don't care about errors, just need skipping, so we can return bool: true to continue, false to skip.

The error propagation can really be done by the caller however they like (e.g. append diagnostics), no need to complicate visit logic.

Reply:
Fair enough, I've updated the function to return a boolean. I flipped the semantics though because skipping on true seemed cleaner.